raphtory-graphql/src/model/graph/graph.rs

```diff
@@ -135,16 +135,118 @@ impl GqlGraph {
             .collect()
     }
 
-    async fn edges<'a>(&self, filter: Option<EdgeFilter>) -> Vec<Edge> {
-        match filter {
-            Some(filter) => self
-                .graph
+    async fn search_count(&self, query: String) -> usize {
+        self.graph.search_count(&query).unwrap_or(0)
+    }
+
+    async fn node_count(&self, filter: Option<NodeFilter>) -> usize {
+        if let Some(filter) = filter {
+            self.graph
+                .vertices()
+                .iter()
+                .map(|vv| vv.into())
+                .filter(|n| filter.matches(n))
+                .count()
+        } else {
+            self.graph.count_vertices()
+        }
+    }
+
+    async fn edge_count(&self, filter: Option<EdgeFilter>) -> usize {
+        if let Some(filter) = filter {
+            self.graph
                 .edges()
                 .into_iter()
                 .map(|ev| ev.into())
                 .filter(|ev| filter.matches(ev))
-                .collect(),
-            None => self.graph.edges().into_iter().map(|ev| ev.into()).collect(),
+                .count()
+        } else {
+            self.graph.count_edges()
+        }
+    }
+
+    async fn edges<'a>(
+        &self,
+        filter: Option<EdgeFilter>,
+        limit: Option<usize>,
+        offset: Option<usize>,
+    ) -> Vec<Edge> {
+        match (filter, limit, offset) {
+            (None, None, None) => {
+                return self.graph.edges().into_iter().map(|ev| ev.into()).collect()
+            }
+            (Some(filter), Some(limit), Some(offset)) => {
+                return get_edges_with_filter_limit(&self.graph, filter, limit, offset);
+            }
+            (Some(filter), None, Some(offset)) => {
+                return get_edges_with_filter_limit(&self.graph, filter, 10, offset)
+            }
+            (Some(filter), Some(limit), None) => {
+                return get_edges_with_filter_limit(&self.graph, filter, limit, 0)
+            }
+            (Some(filter), None, None) => return get_edges_with_filter(&self.graph, filter),
```

Comment:
Probably want to put (None, None, None) (Some, None, None) and (None, Some, Some) and (Some, Some, Some) at the top and the weirder ones lower down.